analytics/js_tracking_api.rst

```diff
@@ -0,0 +1,859 @@
+
+Example of use
+
+.. code-block:: html
+
+    <button onClick = function(){_paq.push(["trackContentInteractionNode", this, "clicked"]);}>Click me!</button>
+
+
+Track interactions and impressions manually
+```````````````````````````````````````````
+If you want to track interactions and impressions fully manually you can use ``trackContentImpression``
+ and ``trackContentInteraction`` (used as a function in eventListener)::
+
+    _paq.push(["trackContentImpression", contentName, contentPiece, contentTarget]);
+
+.. data:: contentName
+
+    **Required** String containing name of Content Impression
+
+.. data:: contentPiece
+
+    **Required** String containing name of Content Impression Piece
+
+.. data:: contentTarget
+
+    **Required** String containing url of Content Impression Target
+
+Example of use::
+
+    _paq.push(["trackContentImpression", "trackingWhitepaper", "document", "http://cooltracker.tr/whitepaper]);
+
+::
+
+    _paq.push(["trackContentInteraction", contentInteraction, contentName, contentPiece, contentTarget]);
+
+.. data:: contentInteraction
+
+    **Required** String containing name of interaction it can be anything ("click" etc). "Unknown" used as default.
+
+.. data:: contentName
+
+    **Required** String containing name of Content Impression
+
+.. data:: contentPiece
+
+    **Required** String containing name of Content Impression Piece
+
+.. data:: contentTarget
+
+    **Required** String containing url of Content Impression Target
+
+Example of use::
+
+    _paq.push(["trackContentImpression", "clicked", "trackingWhitepaper", "document", "http://cooltracker.tr/whitepaper]);
+
+Download and Outlink Tracking
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Tracking Outlink
+````````````````
+
+To enable Download & Outlink tracking run::
+
+    _paq.push(["enableLinkTracking"]);
+
+just after first ``trackPageView`` or ``trackEvent``
+
+.. note::
+
+    All Outlinks are tracked automatically.
+
+To ignore internal outlinks use ``setDomains`` function to define internal domains and subdomains, you can use wildcard::
+
+    _paq(["setDomains", domains]);
+
+.. data:: domains
+
+    **Required** Array with domains written as strings, * are accepted.
+
+Example of usage::
+
+    _paq(["setDomains", ["*.example.com", "*.example.co.uk"]]);
+
+To track clicking a link as an outlink using css class simply add ``piwik_link`` class to link element.
+
+This class name can be changed, use ``setLinkClasses`` to define which CSS class should be tracked::
+
+    _paq.push(["setLinkClasses", className]);
+
+.. data:: className
+
+    **Required** String with css class that should be tracked instead of ``piwik_link``
+
+Example of usage::
+
+    _paq(["setLinkClasses", "track-this-link"]);
+
+If you want to use JS instead you can add ``trackLink`` function to element ``onClick`` attribute::
+
+    _paq.push(["trackLink", linkAddress, "link", dimensions]);
```

Comment:
Why is "link" not a parameter? Does it always have to be this value? 

Update: I see that it's used with different value later on. I would declare it once and link to it when necessary. Another question is: This section describes outlinks, but this function is tracking "link". Is there no distinction between them? In that case section is tracking links and it so happens that by default it tracks only links to external domains.